include/psen_scan_v2/monitoring_frame_deserialization.h

```diff
@@ -88,90 +71,116 @@ class MonitoringFrameFixedFields
   FromTheta from_theta_;
   Resolution resolution_;
 };
-namespace monitoring_frame_additional_field_header_ids
+namespace additional_field
 {
-enum class HeaderID : MonitoringFrameAdditionalFieldHeader::Id
+class Header
+{
+public:
+  using Id = uint8_t;
+  using Length = uint16_t;
+
+public:
+  Header(Id id, Length length);
+
+public:
+  Id id() const;
+  Length length() const;
+
+  static std::string idToString(Id id);
+
+private:
+  Id id_;
+  Length length_;
+};
+
+enum class HeaderID : Header::Id
 {
   SCAN_COUNTER = 0x02,
   DIAGNOSTICS = 0x04,
   MEASURES = 0x05,
   INTENSITIES = 0x06,
   END_OF_FRAME = 0x09
 };
-};  // namespace monitoring_frame_additional_field_header_ids
 
-MonitoringFrameMsg deserializeMonitoringFrame(const MaxSizeRawData& data, const std::size_t& num_bytes);
-MonitoringFrameFixedFields readHeader(std::istringstream& is);
-MonitoringFrameAdditionalFieldHeader readFieldHeader(std::istringstream& is, const std::size_t& max_num_bytes);
-std::vector<MonitoringFrameDiagnosticMessage> deserializeDiagnosticMessages(std::istringstream& is);
+Header read(std::istringstream& is, const std::size_t& max_num_bytes);
+}  // namespace additional_field
 
-class MonitoringFrameFormatError : public std::runtime_error
+Message deserialize(const MaxSizeRawData& data, const std::size_t& num_bytes);
+FixedFields readFixedFields(std::istringstream& is);
+namespace diagnostic
+{
+std::vector<diagnostic::Message> deserializeDiagnosticMessages(std::istringstream& is);
```

Comment:
```suggestion
std::vector<diagnostic::Message> deserializeMessages(std::istringstream& is);
```
Because then we can write:
```
diagnostic::deserializeMessages(...);
```
instead of:
```
diagnostic::deserializeDiagnosticMessages(...);
```